ci/docker-publish.yml

```
@@ -29,11 +29,20 @@ jobs:
     permissions:
       contents: read
       packages: write
+      # This is used to complete the identity challenge
+      # with sigstore/fulcio when running outside of PRs.
+      id-token: write
 
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
 
+      # Install the cosign tool except on PR
+      # https://github.com/sigstore/cosign-installer
+      - name: Install cosign
+        if: github.event_name != 'pull_request'
+        uses: sigstore/cosign-installer@1e95c1de343b5b0c23352d6417ee3e48d5bcd422
```

```suggestion
        uses: sigstore/cosign-installer@main
```

To be consistent with our guidance? Unless the style for these examples requires pinning to a git commit

---

I was trying to be consistent with the other steps, which do this.

---

We require that starter-workflows pin to a specific commit when including actions not maintained by GitHub.  In addition pinning to a ref like main is not recommended as updates could break customers.

---

@chrispat since we are a security-focused action, we recommend using `@main`for the action to adopt fixes ASAP
and to optionally pin a specific release version of the tool itself if that degree of stability is desired

---

As folks involved in supply chain security, we also understand a desire to be very intentional about picking up changes to dependencies 😉

---

Maybe encoding the `@main` recommendation in a comment above that step is a reasonable compromise?

---

You can see what Ruby does: https://github.com/actions/starter-workflows/blob/main/ci/ruby.yml#L29

(I'd encourage you to encourage pinning to a released tag, not `@main`, that's more Actions-idiomatic.)